quickwit-config/src/source_config.rs

```diff
@@ -255,6 +265,37 @@ impl TryFrom<KinesisSourceParamsInner> for KinesisSourceParams {
     }
 }
 
+#[derive(Clone, Debug, PartialEq, Serialize, Deserialize)]
+#[serde(deny_unknown_fields)]
+pub struct LogRotateSourceParams {
+    /// Path of the current log file.
+    #[serde(deserialize_with = "absolute_filepath_from_str")]
+    pub current_file: PathBuf,
```

Comment:
So users are required to pass a `current_file`, and may or may not pass an `archive_dir`?

What happens if `archive_dir` does not contain the file?

Reply:
it should work, reading from `archive_dir` when there are things to read there, and reading `current_file` when there is no longer anything new in `archive_dir`, and going back to `archive_dir` rotation occur. I should probably add a UT to make sure it works as intended. There could be issues however if `archive_dir` is unset, and `name_pattern` would match the filename of `current_file`, which needs to be documented or dealt with